include/utils/hashword.h

```diff
@@ -172,9 +172,12 @@ uint32_t hashword(const uint32_t * k, size_t length, uint32_t initval=0)
   switch(length)                     // all the case statements fall through
     {
     case 3 : c+=k[2];
+      /* FALLTHROUGH */
```

Comment:
Is the compiler parsing this comment somehow? I'm not sure I'm crazy about adding these all over the place...

Reply:
It is.  Search https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html for "fallthrough".

I'm a little dubious of syntactically-meaningful comments too, but the only standard way to suppress these warnings is C++17-only, and I think the comment option is more readable than the C++11 ```[[gnu::fallthrough]]``` extension.

We could selectively disable that warning flag in our compiler options, but I kind of like it, when it works.  It actually caught a bug or two (albeit not in a codepath we ever use...)

Reply:
> the only standard way to suppress these warnings is C++17-only, and I think the comment option is more readable than the C++11 [[gnu::fallthrough]] extension.

Do you also prefer the syntactic comment over `__attribute__ ((fallthrough));`? I think that should be ignored by other compilers, but if it's not, we could always wrap it in a `libmesh_fallthrough();` macro.

Reply:
I prefer the comment, but if we can get a ```libmesh_fallthrough();``` macro to work then I'd be happy with that too, if you'd prefer.  ... and it looks like the macro would probably work as suggested; both the GNU ```__attribute__``` and the C++17 attribute are supposed to be followed by a semicolon.  Should I try that?

Reply:
> Should I try that?

It would be my preference. Aside from "syntactic comments are weird" issue, we don't know that all compilers will eventually support the same comment regexes as GCC (although I don't see why they wouldn't).

Reply:
I'll also write a configure test of C++17 `[[fallthrough]]`. Then the `libmesh_fallthrough();` macro can eventually expand to that on compilers that support the standard attribute specifiers.

Reply:
Sounds good.  We'll hold off on this until I can finish and test (the gcc7 specific part of) that.